app/Exports/LibraryExport.php

```diff
@@ -0,0 +1,58 @@
+<?php
+
+namespace App\Exports;
+
+use APP\Repositories\BookRepository;
+// use DebugBar\DebugBar;
+use Barryvdh\Debugbar\Facades\Debugbar;
+use Illuminate\Support\Collection;
+use Maatwebsite\Excel\Concerns\FromCollection;
+use Maatwebsite\Excel\Concerns\WithHeadings;
+use Maatwebsite\Excel\Concerns\WithTitle;
+use Maatwebsite\Excel\Concerns\ShouldAutoSize;
+use Illuminate\Support\Facades\Log;
+
+class LibraryExport implements FromCollection, WithTitle, WithHeadings, ShouldAutoSize
+{
+    
+    private $bookRepository;
+
+    public function __construct(BookRepository $bookRepository)
+    {
+        $this->bookRepository = $bookRepository;
+    }
+    
+    /**
+    * @return \Illuminate\Support\Collection
+    */
+    public function collection()
+    {
+        $books = $this->bookRepository->all(['*']);
+        $exportData = [];
+        foreach($books as $book) {
+            //possible here to do some data filtering and manipulation etc, but isn't necesary at time of writing
+            $book->makeHidden('updated_at', 'deleted_at', 'created_at');
+            $data = $book->toArray();
+            
+            array_push($exportData, $data);
+        }
+        return new Collection($exportData);
+    }
```

Comment:
With the Laravel Excel package it is common to use a [separate mapping function](https://docs.laravel-excel.com/3.1/exports/mapping.html) in this class that maps a single Book instance to a row in the export. Such mapping function can then be responsible for including / excluding the correct values for each row.

[This](https://gist.github.com/marijnvanderhorst/c1a12bc33d671bb24aef906b942556bc) is an example of what the end result could look like (not tested).